api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImpl.java

```diff
@@ -9,21 +9,36 @@
  */
 package org.openmrs.module.fhir2.api.translators.impl;
 
+import javax.inject.Inject;
+
 import lombok.AccessLevel;
 import lombok.Setter;
 import org.hl7.fhir.r4.model.AllergyIntolerance;
 import org.hl7.fhir.r4.model.CodeableConcept;
 import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Reference;
 import org.openmrs.Allergen;
 import org.openmrs.AllergenType;
 import org.openmrs.Allergy;
 import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.api.FhirPatientService;
+import org.openmrs.module.fhir2.api.dao.FhirUserDao;
 import org.openmrs.module.fhir2.api.translators.AllergyIntoleranceTranslator;
+import org.openmrs.module.fhir2.api.translators.PatientTranslator;
 import org.springframework.stereotype.Component;
 
 @Component
 @Setter(AccessLevel.PACKAGE)
-public class AllergyIntoleranceTranslatorImpl implements AllergyIntoleranceTranslator {
+public class AllergyIntoleranceTranslatorImpl extends AbstractReferenceHandlingTranslator implements AllergyIntoleranceTranslator {
+	
+	@Inject
+	private PatientTranslator patientTranslator;
+	
+	@Inject
+	private FhirPatientService patientService;
```

Comment:
@jecihjoy I was going through your PR to learn how to write better code, I guess that according to https://github.com/openmrs/openmrs-module-fhir2#development-principles we should not use services inside Translators rather we should use Dao. Maybe I am wrong but thought I should point it out

Reply:
@VaishSiddharth this standard is applicable when using OpenMrs services inside translators. However if the service and the translators for the specific resource being accessed in the Translator has already been created, the best approach is to have the fhir service layer calling Dao layer instead of accessing Dao directly. @ibacher can clarify more on this

Reply:
@VaishSiddharth What @jechijoy said here is correct. We’re trying to avoid dependency on the OpenMRS service layer, because my long term vision for this project is to develop a framework that can pulled out of an OpenMRS module and into its own service. I outlined what I think this should look like [in this Talk post](https://talk.openmrs.org/t/the-future-of-openmrs-on-fhir/25865), though there’s a lot of work necessary to make this possible.

Reply:
@jecihjoy @ibacher Thanks for clarifying  :)